mkt/webapps/tasks.py

```diff
@@ -323,13 +323,23 @@ def update_supported_locales(ids, **kw):
 @use_master
 def index_webapps(ids, **kw):
     # DEPRECATED: call WebappIndexer.index_ids directly.
-    WebappIndexer.index_ids(ids, no_delay=True)
+    homescreens = set(
+        Webapp.tags.through.objects.filter(
+            webapp_id__in=ids,
+            tag__tag_text='homescreen')
+        .values_list('webapp_id', flat=True))
+    webapps = set(ids) - homescreens
+    if homescreens:
+        HomescreenIndexer.index_ids(list(homescreens), no_delay=True)
+    if webapps:
+        WebappIndexer.index_ids(list(webapps), no_delay=True)
 
 
 @post_request_task(acks_late=True)
 @use_master
 def unindex_webapps(ids, **kw):
     # DEPRECATED: call WebappIndexer.unindexer directly.
+    HomescreenIndexer.unindexer(ids)
     WebappIndexer.unindexer(ids)
```

Comment:
Can you check if that is covered by tests? If not, add one deleting only a homescreen and another deleting only a webapp ? I'm surprised it's working and does not cause a 404 on elasticsearch and then an exception in our side.